### DISCUSSION TITLE
Unable to handle Failed and stderr error condition in one task - AKEX

### DISCUSSION THREAD

**Discussion by akhilpatwal** [StackStorm/st2#5474] [Q&A]
Hi,

I'm getting below error when my task getting failed. I want to execute workflow, if still the task will get failed.

{
      "type": "error",
      "message": "YaqlEvaluationException: Unable to resolve key 'stderr' in expression '<% result().stderr =~ \"FullyQualifiedErrorId\" or result().return_code != 0 %>' from context.",
      "task_id": "Disable_AD_Account_Email_ID",
      "route": 1,
      "task_transition_id": "ELK_DB_UPDATE_ERROR__t0"
    },
    
    
    
   Code condition that I have given
   
       next:
      - do:
          - SNOW_Failure
          - ELK_DB_UPDATE_ERROR
        publish:
          - subtask_snow: disable_account
          - task_name: Disable AD Account/Email ID
          - fail: Task Failed
        when: <% result().stderr =~ "FullyQualifiedErrorId" or result().return_code != 0 %>
      - do:
          - Manager_Null
        when: <% result().stderr !~ "FullyQualifiedErrorId" %>
        publish:
          - subtask_snow: disable_account
          - task_name: Disable AD Account/Email ID
          - success: Success
      - do:
          - SNOW_Failure
        when: <% failed() %>

**Comment by amanda11** [comment]
From the error it appears that stderr is not an attribute of your result from that action and so that is why the evaluation is failing.

Does your action ever return a stderr object in its result? If not, then amend accordingly.

If it is sometimes present, and sometimes not, then you could change your expression so that you use result.get("stderr") - that way you can specify a default if stderr is not present, see https://yaql.readthedocs.io/en/latest/standard_library.html#get
